examples/models/question_answering/QuestionAnswering.py

```diff
@@ -0,0 +1,495 @@
+        self.topic_year = 2019
+        self.topic_words = ['covid', 'coronavirus disease 19', 'sars cov 2', '2019 ncov', '2019ncov', '2019 n cov',
+                           '2019n cov','ncov 2019', 'n cov 2019', 'coronavirus 2019', 'wuhan pneumonia', 'wuhan virus',
+                           'wuhan coronavirus', 'coronavirus 2', 'covid-19', 'SARS-CoV-2', '2019-nCov']
+        self.df_topic_only = None
+        self.topic_encoded = None
+        self.encoding_model = None
+        self.torch_device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        self.qa_tokenizer = tokenizer 
+
+    def _create_model(self, scratch: bool):
+        qa_model = AutoModelForQuestionAnswering.from_pretrained(BERT_SQUAD);
+        qa_model = qa_model.to(self.torch_device)
+        qa_model.eval()
+        return qa_model
+
+    def train(self, dataset_path=None, shared_params=None, **train_args):
+        pass
+
+    def evaluate(self, dataset_path):
+        return float(1)
+
+
+    def dump_parameters(self):
+        params = {}
+        return params
+
+    def load_parameters(self, params):
+        fine_tune_dataset_path = params
+        print (fine_tune_dataset_path)
+        print (type(fine_tune_dataset_path),'*************')
+        with open(fine_tune_dataset_path, 'rb') as f:
+            zip_file_base64 = f.read()
+
+        with tempfile.NamedTemporaryFile() as tmp:
+            # Convert back to bytes & write to temp file
+            with open(tmp.name, 'wb') as f:
+                f.write(zip_file_base64)
+            with tempfile.TemporaryDirectory() as root_path:
+                dataset_zipfile = zipfile.ZipFile(tmp.name, 'r')
+                dataset_zipfile.extractall(path=root_path)
+                dataset_zipfile.close()
+
+                metadata_path = "{}/metadata.csv".format(root_path)
+                all_json = glob.glob("{}/arxiv/**/*.json".format(root_path), recursive=True)
+
+                self.df_topic_only = DataLoader.json_load_to_df(
+                    all_json=all_json,
+                    metadata_path=metadata_path, topic_words=self.topic_words, topic_year=self.topic_year)
+
+                self.topic_encoded, self.encoding_model = DataLoader.load_to_embedding(self.df_topic_only)
+
+        self._model = self._create_model(scratch=False)
+
+    def predict(self, queries: List[Any]) -> List[Any]:
+        predictions = list()
+        data = queries[0]
+        assert isinstance(data, dict)
+
+        try:
+            task_questions = data['questions']
+        except:
+            task_questions = []
+            for question_set in data.values():
+                task_questions.extend(question_set['questions'])
+
+        dict_queries = []
+        for idx, query in enumerate(task_questions):
+            print(f"Getting answers from query: {query}")
+            answers_list = []
+            if '\n' in query:
+                '''this is for queries with /n separated context'''
```

Comment:
for covid task, is there any context?

Reply:
Covid does not have context. I added this in case user provides the queries with context. When no context given, this part will be skipped.

Reply:
Then you need to put it into the documentation or some other place to explain how to add the context.
Otherwise, only you know that there could be a context field and where to put it.

Reply:
Just added the description into documentation to Singa-Auto pr#80